src/net.cpp

```diff
@@ -2437,6 +2439,7 @@ void CConnman::ThreadOpenConnections(const std::vector<std::string> connect)
             }
             if (!interruptNet.sleep_for(std::chrono::milliseconds(500)))
                 return;
+            PerformReconnections();
```

Comment:
Yes, `ShouldReconnectV1` (which is called in `DisconnectNode()`) will result in the peer only being put on the list of reconnections if it's in `RecvState::KEY`. That is impossible if they ever sent us something, in which case `PerformReconnections()` will do nothing.

Reply:
Is it possible that they haven't sent us anything (the state is `KEY`), but disconnected us for other reason than not supporting `v2`? I assume it's possible if an alternative client implements whatever because this is not prohibited by any bips... But even in the current client, say that `IsBanned` is triggered?

Reply:
Yes, you are right, and I could confirm this with two of my nodes: If we connect via v2 peer and are disconnected because they have banned us, we would try once again with v1 (and then get disconnected again). Both for automatic connections, and (with this PR) manual ones. Although for manual ones it arguably probably matters less because we'll continuously retry with `-connect` and `-addnode` cli args anyway, even when everything is v1.
I don't really see a good way to avoid this, it's not like they notify us why they have disconnected us, so both situations would look the same for us.

Reply:
Okay, so for this PR i'd just suggest clarifying the *newly added* comments like
```            // Attempt v2 connection if we support v2 - if our peer doesn't support it, we'll reconnect with v1.``` 
Right now they are not accurate, as if the banning thing (or some other disconnection reason) don't exist.

If you feel like it, might as well touch the existing comments as well (say around `ShouldReconnectV1`) :)

In future, we might communicate the disconnection reason (Ban, v1, eviction, protocol violation, or whatever else), probably through a new p2p message. Right now I'm not 100% sure the benefits are worthy of code complexity :) Better p2p logging (on the disconnected end) may be even better goal than saving up one useless v1/v2 reconnection.
Curious what you think about this idea.

Reply:
Ok, I clarified the comments added in this PR.
I think that a single connection / reconnection attempt doesn't waste a meaningful amount of resources of either side, so I'm not sure if there is an actual problem here. Also, often one might not want the other side to know that you evicted them because they are banned (after all you banned them for a reason) - that might just help them evade the ban.